### PR TITLE
fix: 1125 issue with unlinking project from qf round

### PR DESCRIPTION
**Pull request by Meriem-BM**
Related to #1125

## Summary by CodeRabbit

- **New Features**
  - Added conditional check to display a message when no projects are found, improving user feedback.

- **Enhancements**
  - Improved data integrity checks during insertion by updating SQL queries.
  - Enhanced handling of sponsor images and banner background images.
  - Added validation for QF round data.

- **Refactor**
  - Refactored logic for sponsor images, banner background images, and QF round validation for better maintainability.

### CODE REVIEW

**Comment by Meriem-BM** [comment]
@mohammadranjbarz can you review this please

File: src/server/adminJs/tabs/qfRoundTab.ts [hunk 1]
```diff
@@ -102,6 +108,80 @@ const returnAllQfRoundDonationAnalysis = async (
...
+  payload.sponsorsImgs = sponsorsImgs;
+  delete payload.totalSponsorsImgs;
+}
+
+async function handleBannerBgImage(payload: any) {
+  if (payload.bannerBgImage && payload.bannerBgImage.path) {
+    const { path, name } = payload.bannerBgImage;
+    const result = await pinFile(fs.createReadStream(path), name);
+    payload.bannerBgImage = `${process.env.PINATA_GATEWAY_ADDRESS}/ipfs/${result.IpfsHash}`;
+  }
```
[the remark this answers is not in the document]
Comment (mohammadranjbarz) [reply src/server/adminJs/tabs/qfRoundTab.ts]: @Meriem-B IT's a good suggestion, can you apply it?

File: src/server/adminJs/tabs/qfRoundTab.ts [hunk 2]
```diff
@@ -102,6 +108,80 @@ const returnAllQfRoundDonationAnalysis = async (
...
+      const result = await pinFile(fs.createReadStream(path), name);
+      sponsorsImgs.push(
+        `${process.env.PINATA_GATEWAY_ADDRESS}/ipfs/${result.IpfsHash}`,
+      );
+      delete payload[`sponsorsImgs.${i}`];
+    }
+  }
+  payload.sponsorsImgs = sponsorsImgs;
+  delete payload.totalSponsorsImgs;
+}
```
[the remark this answers is not in the document]
Comment (mohammadranjbarz) [reply src/server/adminJs/tabs/qfRoundTab.ts]: @Meriem-B  Can you make this change?

**Review by mohammadranjbarz — CHANGES_REQUESTED** [review]
LGTM but will appreciate it if you can make those changes that I mentioned

**Comment by Meriem-BM** [comment]
@mohammadranjbarz done.

**Pull request merged**